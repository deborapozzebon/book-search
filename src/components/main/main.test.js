import { render, screen, fireEvent } from '@testing-library/react';
import Main from './main';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

test('books search', () => {
  render(<Main />);
  const linkElement = screen.getByText(/Books Search/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders without crashing', () => {
  const div = document.createElement("div");
  ReactDOM.render(<Main></Main>, div);
  ReactDOM.unmountComponentAtNode(div);
})

test('input and button', () => {
  render(<Main />);
  const input = screen.getByTestId("search-input");
  const button = screen.getByTestId("search-button");
  expect(input).toBeInTheDocument();
  expect(button).toBeInTheDocument();
});

test('snapshot 2', () => {
  const tree = renderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
})