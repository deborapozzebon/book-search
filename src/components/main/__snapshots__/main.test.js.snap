// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 2 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-light bg-dark"
  >
    <span
      className="navbar-brand"
    >
      Books Search
    </span>
    <div
      className="nav-items navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Search
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Favorites
      </a>
    </div>
  </nav>,
  <footer
    className="footer"
  >
    <div
      className="footer-content"
    />
  </footer>,
  <div
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group mt-2"
      >
        <input
          className="form-control mt-2"
          data-testid="search-input"
          onChange={[Function]}
          placeholder="Book name"
          type="text"
        />
        <button
          className="btn btn-primary mt-2"
          data-testid="search-button"
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
    <ul
      className="pagination"
    >
      <li
        className="previous disabled"
      >
        <a
          aria-disabled="true"
          aria-label="Previous page"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          &lt;
        </a>
      </li>
      <li
        className="next"
      >
        <a
          aria-disabled="false"
          aria-label="Next page"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          &gt;
        </a>
      </li>
    </ul>
  </div>,
]
`;
